ui/raidboss/data/05-shb/alliance/the_puppets_bunker.js

```diff
@@ -1,11 +1,514 @@
+      response: Responses.goSides(),
+    },
+    {
+      id: 'Puppet Compound Mechanical Decapitation',
+      netRegex: NetRegexes.startsUsing({ source: 'The Compound', id: '51B4', capture: false }),
+      response: Responses.getIn(),
+    },
+    {
+      id: 'Puppet Compound Mechanical Contusion',
+      // This is also used for Compound 2P R012: Laser for non-tanks.  It's a spread there too.
+      netRegex: NetRegexes.headMarker({ id: '008B' }),
+      condition: (data, matches) => data.me === matches.target && data.phase === 'compound',
+      response: Responses.spread('info'),
+    },
+    {
+      id: 'Puppet Compound 2P Centrifugal Slice',
+      netRegex: NetRegexes.startsUsing({ source: 'Compound 2P', id: '51B8', capture: false }),
+      condition: Conditions.caresAboutPhysical(),
+      response: Responses.aoe(),
+      // Cover this phase for the checkpoint as well.
+      run: (data) => data.phase = 'compound',
+    },
+    {
+      id: 'Puppet Compound 2P Prime Blade Out',
+      netRegex: NetRegexes.startsUsing({ source: 'Compound 2P', id: ['541F', '5198'], capture: false }),
+      response: Responses.getOut(),
+    },
+    {
+      id: 'Puppet Compound 2P Prime Blade Behind',
+      netRegex: NetRegexes.startsUsing({ source: 'Compound 2P', id: '5420', capture: false }),
+      response: Responses.getBehind(),
+    },
+    {
+      id: 'Puppet Compound 2P Prime Blade In',
+      netRegex: NetRegexes.startsUsing({ source: 'Compound 2P', id: ['5421', '519A'], capture: false }),
+      response: Responses.getIn(),
+    },
+    {
+      id: 'Puppet Compound 2P R012: Laser You',
+      // R012: Laser also puts out 008B headmarkers on non-tanks.
+      netRegex: NetRegexes.headMarker({ id: '00DA' }),
+      condition: Conditions.targetIsYou(),
+      response: Responses.tankBuster('alert'),
+      run: function(data, matches) {
+        data.busterTargets = data.busterTargets || [];
+        data.busterTargets.push(matches.target);
+      },
+    },
+    {
+      id: 'Puppet Compound 2P R012: Laser Not You',
+      netRegex: NetRegexes.headMarker({ id: '00DA', capture: false }),
+      delaySeconds: 0.5,
+      suppressSeconds: 5,
+      alertText: function(data) {
+        if (!data.busterTargets)
+          return;
+        if (data.busterTargets.includes(data.me))
+          return;
+
+        if (data.role === 'healer') {
+          return {
+            en: 'Tank Buster',
+            de: 'Tank buster',
+            fr: 'Tank buster',
+            ja: 'タンクバスター',
+            cn: '坦克死刑',
+            ko: '탱버',
+          };
+        }
+
+        // Note: do not call out "avoid tank" here because there's a lot of markers going out.
+      },
+      run: (data) => delete data.busterTargets,
+    },
+    {
+      id: 'Puppet Compound 2P Three Parts Disdain',
+      netRegex: NetRegexes.headMarker({ id: '003E' }),
+      condition: (data) => data.phase === 'compound',
+      response: Responses.stackOn(),
```

Comment:
Should this include a knockback prevention call too?

Reply:
I am iffy on that.  I don't know that I can count on everybody using knockbacks together, and then the second two hits will hit fewer people as some of the party is not knocked back?  Maybe it's worth calling anyway?  What do you think?

Reply:
I have a feeling that knockback prevention is going to become the standard just because it's an option, but I think you're right that it's worth waiting to see. (This is sort of... parallel, maybe? to the situation with oopsy in Hades EX, where you had to special-case the meteor circles because the standard strategy is to just eat the center one. This Bunker situation is definitely worth just calling the stack for now and seeing where the community takes it.)

Reply:
I had the stack marker and popped knockback prevention; I didn't come close to dying, but I definitely took more damage than usual. I think most people are holding W here so if they didn't get hit on the 2nd stack they'd get hit on the 3rd. I don't have a strong preference here, just that I've seen groups get pushed into the electricity before.

Reply:
Will add this, although I'll have you know that somebody watching my stream hitting knockback replied "I love you just not giving a shit about that knockback stack".  Excuse me, I was in the first hit, and everybody else flew away!!